lib/resty/upstream/healthcheck.lua

```diff
@@ -201,8 +199,14 @@ local function peer_ok(ctx, is_backup, id, peer)
     end
 end
 
+-- shortcut error function for check_peer()
+local function report_error(sock, ctx, is_backup, id, peer, ...)
+  if not peer.down then errlog(...) end
+  sock:close()
```

Comment:
I think it's better to move `sock:close` outside this function, because we don't always need to close the sock.